File: solutions/lua/pascals-triangle/1/pascals-triangle.lua
return function(n)
	local rows = {}
	for row = 1, n do
		local r = {}
		for i = 1, row do
			if i == 1 or i == row then
				r[i] = 1
			else
				r[i] = rows[row - 1][i - 1] + rows[row - 1][i]
			end
		end
		rows[row] = r
	end

	return {
		rows = rows,
		last_row = rows[#rows],
	}
end

File: solutions/ruby/pascals-triangle/1/pascals_triangle.rb
# frozen_string_literal: true

# class Triangle
class Triangle
  attr_reader :rows

  def initialize(total) = @rows = total.times.with_object([]) do |row, rows|
    rows << (row.zero? ? [1] : rows.last.each_cons(2).map(&:sum).prepend(1).push(1))
  end
end
